### PR TITLE
[RW-9193][risk=no]  [UI]: Migrate to use new DisksApi

### PR DESCRIPTION
**Pull request by NehaBroad**
1) Use disksApi rather than diskApi, to support multiple disks for a workspace.
2) Find the jupyter disk from list of Disks by checking if  isGceRuntime=true
3) Rename the diskStore parameter persistentDisk to gcePersistentDisk 

I **have not** added the Rstudio persistentDisk variable yet to diskStore since i am not sure if its being used. 
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/src/testing/stubs/disks-api-stub.ts [hunk 1]
```diff
@@ -30,12 +37,17 @@ export class DiskApiStub extends DiskApi {
...
+    _options?: any
+  ): Promise<ListDisksResponse> {
+    return new Promise<ListDisksResponse>((resolve, reject) => {
       if (!this.disk) {
         reject(Error('disk not found'));
       }
-      resolve(this.disk);
+      const disksResponse: ListDisksResponse = [];
+      disksResponse.push(this.disk);
+      resolve(disksResponse);
```
Comment (jmthibault79) [inline ui/src/testing/stubs/disks-api-stub.ts:50]: why not `resolve([this.disk])` ?

**Review by jmthibault79 — APPROVED** [review]
Everything looks good.  Tested locally.

failing e2e test is https://precisionmedicineinitiative.atlassian.net/browse/RW-9264 so it's OK to merge

**Review by yonghaoy — APPROVED** [review]
Thanks for doing this. 
Just realize that listDiskApi may returns other disks that are not created by creator. 
I have https://github.com/all-of-us/workbench/pull/7221 to fix this. This PR can be merged after that one.

**Comment by jmthibault79** [comment]
I'm rebasing this PR and taking it over while @NehaBroad is out.

**Comment by jmthibault79** [comment]
e2e failures are 2 that we're seeing on other branches (https://pmi-engteam.slack.com/archives/C7JUTPQLE/p1671468452809719) and are unlikely to be related ... merging.

**Pull request merged**